nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -866,6 +863,14 @@ void Costmap2DROS::getCostsCallback(
           pose_transformed.pose.position.x,
         pose_transformed.pose.position.y);
 
+      bool in_bounds = costmap->worldToMap(
+        pose_transformed.pose.position.x,
+        pose_transformed.pose.position.y, mx, my);
+
+      if (!in_bounds) {
+        response->costs.push_back(LETHAL_OBSTACLE);
```

Comment:
In `footprintCostAtPose` if it's out-of-map, LETHAL_OBSTACLE is returned so I changed it from NO_INFORMATION to match. (In my fork I changed all of them to NO_INFORMATION but that's for my use case, I guess it's debatable what it should be)

Reply:
Good catch. Sure, lets keep this.